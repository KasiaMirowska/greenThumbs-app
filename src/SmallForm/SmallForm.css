.small-form {
    /* background-color: cornsilk; */
    color: rgb(43, 90, 43);
    width: 85%;
    margin: auto;

}

.form__field {
    font-family: inherit;
    border: 0;
    border-bottom: 2px solid black;
    outline: 0;
    font-size: 1rem;
    color: grey;
    background: white;  
    padding-left: 5px;
    border-radius: 5px;
    height: 30px;
    margin: auto;
   
  }
  placeholder {
      color: transparent; 
      padding-left: 5px;
    }




  @media (min-width: 600px) {
   
    .small-form {
        width: 85%;
        margin: auto;
        padding-bottom: 5%;
    }
    .form {
        margin: auto;
        margin-left: 50%;
        /* padding: 30px;   */
    }
    
    .form__field {
        width: 40%; 
       margin-left: 3%;
    }
    .button-container {
        margin: auto;
        vertical-align: bottom;
        margin-left: 50%;
    }
}  


   