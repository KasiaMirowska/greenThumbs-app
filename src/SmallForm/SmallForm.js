import React from 'react';
import ProxyCalls from '../Services/ProxyCalls'
import GreenContext from '../Context';
import './SmallForm.css';


export default class SmallForm extends React.Component {
    static contextType = GreenContext;
    constructor() {
        super()
        this.state = {
            error: null,
            //stateSelection: null,
            city: null,
            term: null,
        }
    }
    handleFormSubmit = (e) => {
        e.preventDefault();
        const location = this.state.city;
        const term = this.state.term;

        ProxyCalls.getThroughGreenThumbApi(term, location)
            .then(data => {
                this.context.setList(data.businesses);
                console.log(data)
                this.props.history.push(`/list/${location}`);
            })
            .catch(err => {
                this.setState({
                    error: err
                });
            });
    }

    handleCityInput = e => {
        e.preventDefault();
        const location = e.target.value;
        this.setState({
            city: location,
        });
    }
    handleCategoryInput = e => {
        e.preventDefault();
        console.log(e.target.value)
        this.setState({
            term: e.target.value
        });
    }

    render() {
        
        return (
            <div className='small-form'>

                <form className='form'onSubmit={this.handleFormSubmit}>
                    <div className='select-bar' >
                        <h2>Search for: </h2>
                        
                        <select className="form__field" onChange={this.handleCategoryInput} required>
                            <option value=" ">Select category: </option>
                            <option value="coffee">Coffee-shops</option>
                            <option value="bakery">Bakeries</option>
                            <option value="juice">Juice-Bars</option>
                            <option value="resturants">Restaurants</option>
                            <option value="breakfast">Breakfast</option>
                            <option value="lunch">Lunch</option>
                            <option value="dinner">Dinner</option>
                        </select>
                       
                    {/* </div>


                    <div className='search-bar'> */}
                        <h2>In</h2>
                        <input type='text' className="form__field" placeholder='City' onChange={this.handleCityInput} required />
                    </div>
                    <div className='button-container'>
                        <button>Search</button>
                    </div>
                </form>

            </div>
        );
    }
}