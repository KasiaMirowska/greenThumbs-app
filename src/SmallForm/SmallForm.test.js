import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import SmallForm from './SmallForm';
import {BrowserRouter as BR } from 'react-router-dom';

describe('SmallForm component', () => {
    const props = {
        location: {
            pathname: '/list/:location',
        }
    }
    it ('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDom.render(<BR><SmallForm {...props}/></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
    it('renders UI as expected', () => {
        const form = renderer.create(<BR><SmallForm {...props}/></BR>);
        expect(form.toJSON()).toMatchSnapshot();
    })
})