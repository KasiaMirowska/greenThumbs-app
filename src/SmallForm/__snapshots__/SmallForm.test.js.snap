// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallForm component renders UI as expected 1`] = `
<div
  className="small-form"
>
  <div
    className="error"
  />
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="select-bar"
    >
      <h2>
        Search for: 
      </h2>
      <select
        className="form__field"
        onChange={[Function]}
        required={true}
      >
        <option
          value=" "
        >
          Select category: 
        </option>
        <option
          value="coffee"
        >
          Coffee-shops
        </option>
        <option
          value="bakery"
        >
          Bakeries
        </option>
        <option
          value="juice"
        >
          Juice-Bars
        </option>
        <option
          value="resturants"
        >
          Restaurants
        </option>
        <option
          value="breakfast"
        >
          Breakfast
        </option>
        <option
          value="lunch"
        >
          Lunch
        </option>
        <option
          value="dinner"
        >
          Dinner
        </option>
      </select>
      <h2>
        In
      </h2>
      <input
        className="form__field"
        onChange={[Function]}
        placeholder="City"
        required={true}
        type="text"
      />
    </div>
    <div
      className="button-container"
    >
      <button>
        Search
      </button>
    </div>
  </form>
</div>
`;
